lib/src/main/java/ECSE428Project/service/AccountService.java

```diff
@@ -35,6 +36,29 @@ public Account createAccount(String name, String email, String password) {
         account = accountRepository.save(account);
         return account;
     }
+    
+  @Transactional
+  public Account changePassword(String id, String oldPass, String newPass) {
+    Optional<Account> opt = accountRepository.findById(id);
+    Account account;
+    
+    if (opt.isPresent()) {
+      account = opt.get();
+      
+      //if the oldpassword is correct, we make the change, otherwise the password stays the same
+      if (account.getPassword().equals(oldPass)) {
+        account.setPassword(newPass);
+        account = accountRepository.save(account);
+      }
+
+    } else {
+      //if no account with the id exists, return null
+      return null;
```

Comment:
Same as above
This should maybe throw an error but that scenario should never occur since you probably need to be logged in to your account to change your password which means that the account exists